### DISCUSSION TITLE
Matterbridge running as service - config changes

### DISCUSSION THREAD

**Discussion by nikkilocke** [42wim/matterbridge#1458] [Help / Q&A]
I have matterbridge running as a service on a Linux box. The service control file looks like this:
```
[Unit]
Description=matterbridge
After=network.target
StartLimitBurst=5
StartLimitIntervalSec=10

[Service]
ExecStart=/opt/matterbridge/matterbridge -conf /opt/matterbridge/matterbridge.toml
Restart=always

[Install]
WantedBy=multi-user.target
```
Someone set this up for me, and then disappeared. I have just noticed there is a cron job to restart the matterbridge service every hour, and I'm not sure why this is.

If I change the matterbridge.toml file (e.g. to add a new gateway), will matterbridge reload the new file and start the new gateway automatically? Or do I have to stop and restart it? (If the latter, perhaps that's the reason for the cron job?).

I am getting some duplicated messages, but I am also getting some error messages 
`error:websocket: close 1006 (abnormal closure): unexpected EOF" prefix=matterclient`
`download failed: : Unable to get the file info., " prefix=mattermost`
`download failed: File "file_1578.mp4" to large to download (11446756). MediaDownloadSize is 1000000" prefix=telegram`

Would the error message explain the duplicates? 
Or might it be the stopping and restarting?

If it might be the stopping and restarting, is there a proper way of stopping it, so it completes the notifications that the telegram messages have been seen?

**Comment by qaisjp** [comment]
> If I change the matterbridge.toml file (e.g. to add a new gateway), will matterbridge reload the new file and start the new gateway automatically? Or do I have to stop and restart it? (If the latter, perhaps that's the reason for the cron job?).

Auto-reload depends on the setting. Usually I restart matterbridge after each update unless it's a small update that I know works with the auto-reload. Iirc, gateway additions aren't fully supported everywhere.

I would just get rid of the cron job and only restart when you need it to.

> `download failed: File "file_1578.mp4" to large to download (11446756). MediaDownloadSize is 1000000" prefix=telegram`

this is not an error

> `error:websocket: close 1006 (abnormal closure): unexpected EOF" prefix=matterclient`

I don't know for sure, but I would expect matterclient to gracefully reconnect here. I think this can just happen if a connection gets dropped.

> If it might be the stopping and restarting, is there a proper way of stopping it, so it completes the notifications that the telegram messages have been seen?

I'm not sure what you mean

**Comment by nikkilocke** [comment]
Thanks for your helpful response.

> > If it might be the stopping and restarting, is there a proper way of stopping it, so it completes the notifications that the telegram messages have been seen?

>I'm not sure what you mean

As I understand it, the telegram bot will keep sending the same message again and again until it receives a request with a later message number. So duplicate messages would happen if the bridge was stopped between receiving a message and asking for more messages with a later message number.

Is there a way of stopping the bridge gracefully, so that we can be sure it has processed all the messages in the queue, and told telegram not to send them again?

**Reply by minecraftchest1** [reply]
I have never seen that happen Just stop it like you would any other command line app.
